tests/monitoring/monitoring_service/test_service.py

```diff
@@ -0,0 +1,25 @@
+from unittest.mock import Mock, patch
+
+from monitoring_service.service import MonitoringService
+from raiden.tests.utils.factories import make_transaction_hash
+
+
+def test_check_pending_transactions(web3, wait_for_blocks, monitoring_service: MonitoringService):
+    required_confirmations = 3
+
+    monitoring_service.context.required_confirmations = required_confirmations
+    monitoring_service.database.add_waiting_transaction(waiting_tx_hash=make_transaction_hash())
+
+    for tx_status in (0, 1):
+        tx_receipt = {"blockNumber": web3.eth.blockNumber, "status": tx_status}
+        with patch.object(web3.eth, "getTransactionReceipt", Mock(return_value=tx_receipt)):
+            with patch.object(
```

Comment:
You can use multiple context managers with a single `with`, which reduced the nesting.